mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java

```diff
@@ -701,8 +701,48 @@ private void writeHttpResponse(Socket socket, BufferedSink sink, MockResponse re
 
     Buffer body = response.getBody();
     if (body == null) return;
-    sleepIfDelayed(response);
-    throttledTransfer(response, socket, body, sink, Long.MAX_VALUE);
+    if (response.getSocketPolicy() == SocketPolicy.STREAM_REPEAT_FINAL_CHUNK) {
```

Comment:
.. . . can't you accomplish the same with the existing API, using throttled transfer and a sufficiently large response?